assets/src/dashboard/app/font/fontProvider.js

```diff
@@ -57,15 +49,26 @@ function FontProvider({ children }) {
 
   useEffect(() => {
     let mounted = true;
-    getAllFonts().then((newFonts) => {
-      if (mounted) {
-        setFonts(newFonts);
-      }
-    });
+    async function loadFonts() {
+      const newFonts = await import(
+        /* webpackChunkName: "chunk-fonts" */ '../../../fonts/fonts'
+      );
+      const formattedFonts = newFonts.default.map((font) => ({
+        name: font.family,
+        value: font.family,
+        ...font,
+      }));
+
+      setFonts(formattedFonts);
```

Comment:
This call however needs to look at `mounted` (if anything) - only if it's still true, should it set the fonts. Otherwise, it might technically update a stale component.

However given that this provider is effectively a singleton, that'll always be present in the dashboard (will it though? not that familiar with dashboard architecture? ping @carlos-kelly), you might scrap the `mounted` variable completely, as it's not relevant anymore.

One consideration would though be to **not** load the fonts on the first render, but only the first time they're requested. That would require some reworking here though.

Reply:
I think I need to defer to @carlos-kelly and @BrittanyIRL on this one. My ticket is just remove the fonts api and load via JSON. But there is little some working that needs to happen to both dashboard and editor after this change goes in.

Reply:
Yeah! So the reason for `mounted` was in some cases the user would navigate away from the preview of a story before the fonts loaded so it would then try to set those fonts on the provider for a component that was no longer in the tree.

Reply:
So now that fonts are being loaded via json file, this should be much faster. Is this `mounted` still required?